src/header/header.js

```diff
@@ -30,69 +29,79 @@ class BULHeader extends LitElement {
-            <strong>My Account</strong>
-            <select>
-              <option>Option 1</option>
-              <option>Option 2</option>
-            </select>
+          <div class="flex-end primary-nav-right phm">
+            <slot name="primary-nav-right">
+              Library<bulib-libsel></bulib-libsel>
+            </slot>
           </div>
         </div>
-        <div class="secondary-nav">
-          <div id="secondary-nav-main" class="inline pal">${secondaryNavMain}</div>
-          <div id="secondary-nav-search" class="inline">
-            <bulib-search str_default="${this.curr_search}" str_options="${this.str_options}"></bulib-search>
+        <div class="secondary-navbar">
+          <div class="secondary-nav-left pam">
+            <slot name="secondary-nav-left">
+              <strong>${this.curr_secondary}</strong>
+            </slot>
+          </div>
+          <div class="secondary-nav-main pam">
+            <slot name="secondary-nav-main"></slot>
+          </div>
+          <div class="secondary-nav-right pam">
+            <bulib-search str_selected="${this.curr_search}" str_options="${this.str_options}"></bulib-search>
           </div>
         </div>
       </nav>`;
   }
 
-  /** once html is on the page, add classes based on 'curr_*' values */
-  updated(){
-    // set primary nav 'active' styling
-    let i, li;
-    let lsListItems = document.getElementById("site-links").getElementsByTagName("li");
+  /** for production purposes, react to url updating when the component is first loaded */
+  connectedCallback(){ 
+    if(!local) { this._urlUpdated(); } 
+  }
+
+  /** for development purposes, react to manual changes to `this.curr_url` via _urlUpdated */
+  attributeChangedCallback(name, oldValue, newValue){
+    super.attributeChangedCallback(name, oldValue, newValue);
+    if(debug){ console.log(`bulib-header) attributeChangedCallback() called, changing '${name}' from '${oldValue}' to '${newValue}'...`); }
+    switch(name){
+      case "curr_url":      this._urlUpdated(); break;
+      case "curr_primary":  this._primaryUpdated(); break;
+      default: break;
+    }
+  }
+  
+  /** set primary nav 'active' styling */
+  _primaryUpdated(){
+    let i, li; 
+    let lsListItems = this.shadowRoot.querySelector("#site-links").getElementsByTagName("li");
     for(i = 0; i<lsListItems.length; i++) {
       li = lsListItems[i];
-      if((li.id).includes(this.curr_primary)){ li.classList.add("active"); }
+      if((li.id).includes(this.curr_primary)){ li.classList.add("active"); } 
       else{ li.classList.remove("active"); }
     }
   }
 
-  /** update current properties to inform what to display */
-  _setCurrSiteInfo(){
-    let currentUrl = (this.curr_url)? this.curr_url : window.location.href;
-
-    this.curr_library = "";
+  /** set the primary, secondary, and search information according to the currentUrl  */
+  _urlUpdated(){
+    let currentUrl = (local)? this.curr_url : window.location.href;
```

Comment:
refactor this so that the url-specific stuff (determining what `curr_primary` should be) is contained and the actual setting of the primary and secondary data is done on its own.

Reply:
this should be a future optimization after the first release.